### PR TITLE
docs: refactor the developing locally page around abctl

### PR DESCRIPTION
**Pull request by bgroff**
## What
Rework the developing on Airbyte guide to use abctl as the replacement for docker compose.

## Can this PR be safely reverted and rolled back?
- [X] YES 💚

### CODE REVIEW

File: docs/contributing-to-airbyte/developing-locally.md
Comment (davinchia) [inline docs/contributing-to-airbyte/developing-locally.md:26]: @alafanechere should this be 3.10?

File: docusaurus/sidebars.js
Comment (davinchia) [inline docusaurus/sidebars.js:384]: what happened to this file? should it also be deleted?

File: docs/contributing-to-airbyte/resources/developing-locally.md
Comment (davinchia) [inline docs/contributing-to-airbyte/resources/developing-locally.md:105]: is there an abctl equivalent to this section?

File: docs/contributing-to-airbyte/developing-locally.md
Comment (davinchia) [inline docs/contributing-to-airbyte/developing-locally.md:1]: I mostly skimmed this looking for abctl replacements in the new docs. Lmk if there are specific pieces you want a more thorough review of.

File: docs/contributing-to-airbyte/developing-locally.md
Comment (alafanechere) [reply docs/contributing-to-airbyte/developing-locally.md:26]: Yes it should

File: docs/contributing-to-airbyte/developing-locally.md
Comment (evantahler) [inline docs/contributing-to-airbyte/developing-locally.md:4]: [nit]
```suggestion
is largely done in Python by community contributors, though sometimes Java is used for performance reasons. Platform development is done in Java 
```

File: docs/contributing-to-airbyte/developing-locally.md
Comment (evantahler) [inline docs/contributing-to-airbyte/developing-locally.md:52]: Can you provide an example command & output, e.g. `docker images | grep airbyte/source-<source-name>`

Sometimes airbyte is the first docker project folks will have encountered.

File: docs/contributing-to-airbyte/developing-locally.md
Comment (evantahler) [inline docs/contributing-to-airbyte/developing-locally.md:58]: So that's how you do it!

@alafanechere - perhaps we could get a new flag to airbyte-ci to do this automatically, e.g. `airbyte-ci connectors --name source-<source-name> --abctl-load build`.  Not a blocking part of this PR

File: docs/contributing-to-airbyte/developing-locally.md
Comment (evantahler) [inline docs/contributing-to-airbyte/developing-locally.md:62]: [nit] 
```suggestion
The above connector image is tagged with `dev`. You can change this to use another tag if you'd like via the `docker image tag` command.
```

File: docs/contributing-to-airbyte/developing-locally.md
Comment (evantahler) [inline docs/contributing-to-airbyte/developing-locally.md:88]: cc @pedroslopez - I thought we fixed this recently (https://github.com/airbytehq/airbyte-platform-internal/pull/9791)

File: docs/contributing-to-airbyte/developing-locally.md
Comment (evantahler) [inline docs/contributing-to-airbyte/developing-locally.md:112]:
```suggestion
`./gradlew build` creates all the necessary artifacts \(Webapp, Jars, and Docker images\) so that you can run Airbyte locally. Since this builds everything, it can take some time.
```

**Review by evantahler — COMMENTED** [review]
This is great - thank you @bgroff!

I added some nits.  The main suggestions I have are:
1. I think the `Connector Specification Caching` section isn't needed any more
2. You might want to describe (or reference another page) where you describe what `kind` is why/how we use it.  The commands to load dev images all involve kind, so if folks need to troubleshoot, we should point them in the right direction

File: docs/contributing-to-airbyte/developing-locally.md
Comment (alafanechere) [reply docs/contributing-to-airbyte/developing-locally.md:58]: I think it's what @perangel is currently working on.

File: docs/contributing-to-airbyte/developing-locally.md
Comment (pedroslopez) [reply docs/contributing-to-airbyte/developing-locally.md:88]: @evantahler we did fix the issue where the `dev` spec wasn't refetched, but as mentioned on the PR:

> This still requires users to go to the settings page and press the "update" button to re-fetch the spec (it will not dynamically update).

So the instructions here are correct in that you'll need to go to the settings page and click the update button, _however_ you shouldn't need to move to a different version and then back to dev for it to work. (the  "Update the version number to most recent release docker image tag, e.g. `1.6.4`" step shouldn't be necessary)

File: docusaurus/sidebars.js
Comment (bgroff) [reply docusaurus/sidebars.js:384]: The developing-locally moved up a directory. The developing-on-docker is still there in case someone had a link. It did not contain accurate information, I was planning leaving it, but I have delete it.

File: docs/contributing-to-airbyte/resources/developing-locally.md
Comment (bgroff) [reply docs/contributing-to-airbyte/resources/developing-locally.md:105]: Not yet, it is on the abctl board though and we hope to get to it in the next ~week. Will update the documentation then.

File: docs/contributing-to-airbyte/developing-locally.md
Comment (bgroff) [reply docs/contributing-to-airbyte/developing-locally.md:58]: We have put support for publishing directly on hold for now. This can be added at some point once we figure out if we are going to bring a registry with us or not.

File: docs/contributing-to-airbyte/resources/developing-locally.md
Comment (davinchia) [reply docs/contributing-to-airbyte/resources/developing-locally.md:105]: Sounds good. Mainly for the connector teams, so non-blocking for me.

**Comment by davinchia** [comment]
@bgroff after playing around with abctl today, I echo @evantahler's sentiments.

I think we can start with something basic to keep things light. 

Basically,
- couple of sentences on what Kind is.
- how to access the underlying instances - probably a couple of kubectl commands with the kubeconfig specified is sufficient. Basically some of my debugging questions today. Could even link to existing kubernetes docs for conciseness.
  - the workflows I can think of to start: viewing logs, restarting a pod, describing a pod.

All non-blocking for me, can follow up in PRs if they will be fairly quick-coming. Will defer to Evan if he feels stronger and wants to block.

File: docs/contributing-to-airbyte/developing-locally.md
Comment (davinchia) [inline docs/contributing-to-airbyte/developing-locally.md:46]: nice!

**Review by davinchia — APPROVED** [review]
:shipit:

**Pull request merged**
